### PR TITLE
Add workflow to publish binaries

### PR DESCRIPTION
**Pull request by wojciechos**
The workflow will build and publish artifacts with binaries for Linux, Darwin, and Windows for every tag push with a version to the repository.

### CODE REVIEW

File: .github/workflows/build-binaries.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: Build Binaries
+
+on:
+  push:
+    tags:
+      - 'v*.*.*'
+  workflow_dispatch:
```
Comment (IronGauntlets) [inline .github/workflows/build-binaries.yml:7]: I am assuming this workflow would run every time a new tag is pushed or updated?

**Review by IronGauntlets — APPROVED** [review]

File: .github/workflows/build-binaries.yml (hunk 1 above)
Comment (wojciechos) [reply .github/workflows/build-binaries.yml]: just changed to only '*'  - but yeah, the workflow should be triggered when any new tag is pushed

Comment (wojciechos) [reply .github/workflows/build-binaries.yml]: ...or could be triggered manually

Comment (IronGauntlets) [reply .github/workflows/build-binaries.yml]: I don't think we should build on all tags. Should keep the regex at `v*`?

Comment (wojciechos) [reply .github/workflows/build-binaries.yml]: yeah, we can try with only that for now - changed

**Comment by omerfirmak** [comment]
Does it make sense to run this on releases only ? like; https://github.com/bosagora/agora/blob/v0.x.x/.github/workflows/release.yaml#L14

**Review by joshklop — APPROVED** [review]

**Comment by wojciechos** [comment]
> Does it make sense to run this on releases only ? like; https://github.com/bosagora/agora/blob/v0.x.x/.github/workflows/release.yaml#L14

Yeah, that make sense but maybe we can wait with that for now as I have to test each binary manually before releasing it.

**Pull request merged**